src/ngamsServer/ngamsServer/ngamsDaemon.py

```diff
@@ -51,7 +51,7 @@ def start(args, cfg, pidfile):
         return 1
 
     # Go, go, go!
-    with daemon.DaemonContext(pidfile=lockfile.pidlockfile.PIDLockFile(pidfile, timeout=1)):
+    with daemon.DaemonContext(pidfile=lockfile.pidlockfile.PIDLockFile(pidfile)):
```

Comment:
That's unfortunate, as the timeout mechanism is useful to avoid infinite hangs at startup. I'll take this as-is, but will change later to react differently depending on the python-daemon version, adding the extra `timeout` argument when possible (e.g., not when using 1.6).

Reply:
Mmmm, looking more closely this seems a problem with the version of the `lockfile` package rather than the `python-daemon` one. Which versions of lockfile and python-daemon are you using? All lockfile versions I could find in PyPI had the `timeout` argument there (except for 0.8, but that didn't have the PIDLockFile class to begin with).

In any case the extra checks I was thinking to introduce would work similarly, I'm just double-checking they are actually required.

Reply:
Centos7 comes with python-lockfile version 0.9.1.
What do you mean "to avoid infinite hangs at startup"? I haven't seen this problem yet. I assume that the ngamsDaemon hangs but what do you mean by infinite?

Reply:
Wrong assumption again from my side, I think: if the PID lock file is already taken, I thought the the PIDLockFile class would wait indefinitely until the file became available if no timeout was given, but it's actually more likely that it fails immediately. Having an explicit timeout gives the class the change to wait a bit before giving up.

Reply:
OK, I will double check. Perhaps I am mistaken. Thank you for the feedback.

Reply:
I get the following stack trace...

Traceback (most recent call last):
  File "/usr/bin/ngamsDaemon", line 9, in <module>
    load_entry_point('ngamsServer==11.0.2', 'console_scripts', 'ngamsDaemon')()
  File "/usr/lib/python2.7/site-packages/ngamsServer/ngamsDaemon.py", line 174, in main
    exitCode = start(args, cfg, pidfile)
  File "/usr/lib/python2.7/site-packages/ngamsServer/ngamsDaemon.py", line 54, in start
    with daemon.DaemonContext(pidfile=lockfile.pidlockfile.PIDLockFile(pidfile, timeout=1)):
TypeError: __init__() got an unexpected keyword argument 'timeout'

Reply:
I think I will try and create packages for newer versions of python-daemon and python-lockfile. This will solve the problem

Reply:
Weird; the argument is in lockfile 0.9.1 from PyPI, but clearly isn't in the package for the same version provided by CentOS (thanks for double-checking!). I'll take your patch as-is then, and on top of that I will add the option to pass down or not the timeout argument as previously mentioned (basically using the method I used in the command line example above). Considering that, you might not want to create the new packages for python-daemon and python-lockfile.